WordPress/Classes/ViewRelated/Post/WPPostViewController.m

```diff
@@ -1946,6 +1946,47 @@ - (void)editorViewController:(WPEditorViewController *)editorViewController vide
     [self refreshNavigationBarButtons:NO];
 }
 
+- (void)editorViewController:(WPEditorViewController *)editorViewController imagePasted:(UIImage *)image
+{
+    MediaService *mediaService = [[MediaService alloc] initWithManagedObjectContext:[[ContextManager sharedInstance] mainContext]];
+    __weak __typeof__(self) weakSelf = self;
+    NSString *mediaUniqueID = [self uniqueIdForMedia];
+    NSProgress *createMediaProgress = [[NSProgress alloc] initWithParent:nil userInfo:nil];
+    createMediaProgress.totalUnitCount = 2;
+    
+    [self trackMediaWithId:mediaUniqueID usingProgress:createMediaProgress];
+    [mediaService createMediaWithImage:image
+                           withMediaID:mediaUniqueID
+                       forPostObjectID:self.post.objectID
+                     thumbnailCallback:^(NSURL *thumbnailURL) {
+                         __typeof__(self) strongSelf = weakSelf;
+                         if (!strongSelf) {
+                             return;
+                         }
+                         [[NSOperationQueue mainQueue] addOperationWithBlock:^{
+                             [strongSelf.editorView insertLocalImage:thumbnailURL.path uniqueId:mediaUniqueID];
+                         }];
+                     }
+                            completion:^(Media *media, NSError *error) {
+                                __typeof__(self) strongSelf = weakSelf;
+                                if (!strongSelf) {
+                                    return;
+                                }
+                                createMediaProgress.completedUnitCount++;
+                                if (error || !media || !media.absoluteLocalURL) {
+                                    [strongSelf stopTrackingProgressOfMediaWithId:mediaUniqueID];
+                                    [WPError showAlertWithTitle:NSLocalizedString(@"Failed to paste image",
```

Comment:
@nheagy we have a script that updates the strings file which we then upload to get translated before each release.